packages/jellyfish-transaction-builder/src/txn/txn_builder.ts

```diff
@@ -184,3 +185,31 @@ interface Prevouts {
   vin: Vin[]
   total: BigNumber
 }
+
+/**
+ * @param {Prevout[]} prevouts to join
+ * @param {BigNumber} minSum minimum sum expected
+ * @return {Prevouts}
+ */
+function joinPrevoutsForMinSum (prevouts: Prevout[], minSum: BigNumber): Prevouts {
+  const prevOutsNew: Prevout[] = []
+  const vin: Vin[] = []
+  let total = new BigNumber(0)
+
+  for (const prevout of prevouts) {
+    if (total.isGreaterThan(minSum)) {
+      break
+    }
+
+    total = total.plus(prevout.value)
+    prevOutsNew.push(prevout)
+    vin.push({
+      txid: prevout.txid,
+      index: prevout.vout,
+      script: { stack: [] },
+      sequence: 0xffffffff
+    })
+  }
+
+  return { prevouts: prevOutsNew, vin, total }
```

Comment:
On https://github.com/DeFiCh/jellyfish/blob/4cb0d9891432be262983c7a7c55770c9318e2434/packages/jellyfish-transaction-builder/src/provider.ts#L12-L28

There is this that moves the "selecting Prevout" logic to the Provider. However if we do it here, we might compete the abstraction and making it non agnoistic.

I think it's probably better to put the fix logic at https://github.com/DeFiCh/jellyfish/blob/4cb0d9891432be262983c7a7c55770c9318e2434/packages/jellyfish-transaction-builder/__tests__/provider.mock.ts#L23 

so that it can stay implementation agnostic?